### PR TITLE
Add saveProfilesToGcs option; deprecate enableProfilingAgent

### PR DESCRIPTION
**Pull request by bjchambers**

### CODE REVIEW

**Comment by davorbonaci** [comment]
Is this a backport?

File: sdk/src/main/java/com/google/cloud/dataflow/sdk/runners/DataflowPipelineRunner.java [hunk 1]
```diff
@@ -275,6 +275,14 @@ public static DataflowPipelineRunner fromOptions(PipelineOptions options) {
     if (dataflowOptions.getTempLocation() != null) {
       validator.validateOutputFilePrefixSupported(dataflowOptions.getTempLocation());
     }
+    if (!Strings.isNullOrEmpty(dataflowOptions.getSaveProfilesToGcs())) {
+      validator.validateOutputFilePrefixSupported(dataflowOptions.getSaveProfilesToGcs());
+    }
+    if (dataflowOptions.getEnableProfilingAgent()) {
+      LOG.error("--enableProfilingAgent is no longer supported, and will be ignored. "
+          + "Use --saveProfilesToGcs instead");
```
Comment (davorbonaci) [inline sdk/src/main/java/com/google/cloud/dataflow/sdk/runners/DataflowPipelineRunner.java]: '.' at the end of the sentence ;-)

**Comment by bjchambers** [comment]
Not really backport or forward port. Just wrote both PRs. See https://github.com/apache/incubator-beam/pull/1554.

File: sdk/src/main/java/com/google/cloud/dataflow/sdk/runners/DataflowPipelineRunner.java (hunk 1 above)
Comment (bjchambers) [inline sdk/src/main/java/com/google/cloud/dataflow/sdk/runners/DataflowPipelineRunner.java]: Done.

**Comment by bjchambers** [comment]
Travis failures seem spurious -- WordCount class is missing. I'll squash and force push to kick off new run.

**Comment by bjchambers** [comment]
This is back to failing with:

```
[WARNING] 
java.lang.ClassNotFoundException: com.google.cloud.dataflow.examples.WordCount
	at java.net.URLClassLoader$1.run(URLClassLoader.java:372)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:361)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.net.URLClassLoader.findClass(URLClassLoader.java:360)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	at org.codehaus.mojo.exec.ExecJavaMojo$1.run(ExecJavaMojo.java:281)
```

**Pull request merged**